### ISSUE TITLE
Team-based collective interaction with deprecated active-set collectives

### ISSUE THREAD

**Issue by naveen-rn** [openshmem-org/specification#303]
I suppose, the interaction between deprecated active-set based collectives (OpenSHMEM-1.5) and team-based collectives are undefined. Concurrent collective operations from different teams and a active-set based collective is incorrect. Please clarify.

**Comment by nspark** [comment]
Since this might catch the eyes of a wider audience, I'll repeat my answer here:

I think your code example exhibits undefined behavior. My rationale is that `shmem_team_barrier` includes the following note:

> The `shmem_barrier_all` routine is equivalent to calling `shmem_ctx_quiet` on the default context followed by calling `shmem_team_sync` on the default team.

Because of this equivalence, I think your code example is undefined due to the concurrent use of the default/world team.

That said, because there's no explicit connection between active-set collectives and team collectives, I think the following code _should_ have well-defined behavior (assuming proper initialization of the `pSync` array):

```c
#pragma omp parallel num_threads(2)
{
    if (omp_get_thread_num() == 0) {
        shmem_team_barrier(SHMEM_TEAM_WORLD);
    } else if (omp_get_thread_num() == 1) {
        shmem_barrier(0, 0, shmem_n_pes(), pSync);
    }
}
```

**Comment by jdinan** [comment]
@naveen-rn @nspark Are we happy with the current text on this? Plus, the extra mention in Annex C that @davidozog is working on?

**Comment by nspark** [comment]
I think we could revise the _Collectives_ introduction as follows (**added**/~~removed~~ text):

> ### Team-implicit collectives
>
> **Some OpenSHMEM collective routines implicitly operate on the world team. These routines include:**
>
> - ~~The~~ `shmem_sync_all` ~~routine~~, **which** synchronizes all PEs in the computation through the default team. This routine is equivalent to a call to `shmem_team_sync` on the world team.
> - ~~The~~ `shmem_barrier_all` ~~routine~~, **which** synchronizes all PEs in the default team and ensures completion of all local and remote memory updates issued via the default context. This routine is equivalent to a call to `shmem_ctx_quiet` on the default context followed by a call to `shmem_team_sync` on the default team.
> - **OpenSHMEM memory management routines, which imply one or more calls to a routine equivalent to `shmem_barrier_all`.**

Still, this doesn't actually address the implicit, disjoint nature of (internal) resources between teams and active sets. I'm not sure I want to actually add something saying "it's safe to use active-set-based collectives concurrently with team-based collectives in multiple threads...", since that's not exactly a good practice.

How does this sound @davidozog @jdinan @naveen-rn?

**Comment by jdinan** [comment]
Sounds good to me. Per #371 please use "world team" instead of "default team".

**Comment by nspark** [comment]
Closed by nspark/specification#20
